docs/development/vsphere-dev.md

```diff
@@ -0,0 +1,125 @@
+
+### Setup CoreDNS server
+1. Login to vSphere Client.
+2. Right-Click on ESX host on which you want to deploy the DNS server.
+3. Select Deploy OVF template.
+4. Copy and paste URL for [OVA](https://storage.googleapis.com/kubernetes-anywhere-for-vsphere-cna-storage/coredns.ova).
+5. Follow next steps according to instructions mentioned in wizard.
+6. Power on the imported VM.
+7. SSH into the VM and execute ./start-dns.sh under /root. Username/Password: root/kubernetes
+
+### Check DNS server is ready
+On your local machine, execute the following command:
+```bash
+dig @[DNS server's IP] -p 53 NS skydns.local
+```
+
+Successful answer should look like the following:
+```bash
+; <<>> DiG 9.8.3-P1 <<>> @10.162.17.161 -p 53 NS skydns.local
+; (1 server found)
+;; global options: +cmd
+;; Got answer:
+;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 42011
+;; flags: qr aa rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1
+
+;; QUESTION SECTION:
+;skydns.local.			IN	NS
+
+;; ANSWER SECTION:
+skydns.local.		160	IN	NS	ns1.ns.dns.skydns.local.
+
+;; ADDITIONAL SECTION:
+ns1.ns.dns.skydns.local. 160	IN	A	192.168.0.1
+
+;; Query time: 74 msec
+;; SERVER: 10.162.17.161#53(10.162.17.161)
+;; WHEN: Tue Mar 14 22:40:06 2017
+;; MSG SIZE  rcvd: 71
+```
+
+### Add DNS server information when create cluster
+Add ```--dns=private --vsphere-coredns-server=http://[DNS server's IP]:2379``` into the ```kops create cluster``` command line.
+
+### Use CoreDNS supported DNS Controller
+Information about DNS Controller can be found [here](https://github.com/kubernetes/kops/blob/master/dns-controller/README.md)
+Currently the DNS Controller is an add-on container and the image is from kope/dns-controller.
+Before the vSphere support is officially merged into upstream, we need to set up CoreDNS supported DNS controller manually.
+```bash
+DOCKER_REGISTRY=[your docker hub repo] make dns-controller-push
+export VSPHERE_DNSCONTROLLER_IMAGE=[your docker hub repo]
+make
+kops create cluster ...
+```
+
+## Kops with vSphere
+vSphere cloud provider support in kops is a work in progress. To try out deploying kubernetes cluster on vSphere using kops, some extra steps are required.
+
+### Pre-requisites
++ vSphere with at least one ESX, having sufficient free disk space on attached datastore. ESX VM's should have internet connectivity.
++ Setup DNS following steps given in relevant Section above.
++ Create the VM using this template (TBD).
++ Currently vSphere code is using AWS S3 for storing all configurations, specs, addon yamls, etc. You need valid AWS credentials to try out kops on vSphere. s3://your-objectstore/cluster1.skydns.local folder will have all necessary configuration, spec, addons, etc., required to configure kubernetes cluster. (If you don't know how to setup aws, then read more on kops and how to deploy a cluster using kops on aws)
```

Comment:
The user-defined s3 storage PR will directly go to kops upstream. We may need to keep this one before that PR getting merged.